### PR TITLE
Fix: Add Array support to Action Server UI action run view

### PR DESCRIPTION
**Pull request by rihardsgravis**
- Add initial Array support to Action Server UI action run view
- Update the Action Run form to hold actual payload value types, not strings

image

### CODE REVIEW

File: action_server/frontend/src/lib/formData.ts [hunk 1]
```diff
@@ -2,29 +2,34 @@
 import type { OpenAPIV3_1 } from 'openapi-types';
 import { InputProperty } from './types';
 
+export type PropertyFormDataType =
+  | string
+  | number
+  | boolean
+  | object
+  | Array<string | number | boolean>;
```
Comment (fabioz) [inline action_server/frontend/src/lib/formData.ts]: Question: Does this mean that something as `Array<object>` or `Array<Array...>>` is not supported?

We can probably merge even if not (and create an issue to be handled in the future -- I guess we could even have  a "json" fallback for more complex cases).

**Review by fabioz — APPROVED** [review]

**Comment by fabioz** [comment]
p.s.: Please add entry to changelog.md.

File: action_server/frontend/src/lib/formData.ts (hunk 1 above)
Comment (rihardsgravis) [reply action_server/frontend/src/lib/formData.ts]: Good catch! Fixed the type which was a bit incorrect, but `Array<object>` and `Array<Array...>>` are supported -  the only note that an `object` is not really a data type in OpenAPI – an object would always be expanded as the `properties` of that object.

For the JSON payload input I added a ticket https://github.com/robocorp/robocorp/issues/310, could add that this week at some point.

File: action_server/docs/CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,7 @@
 ## 0.1.4 - 2024-03-20
 
 - Fixed issue in action-server binary build. 
+- Add support parsing Array type in Action Server UI action run view
```
Comment (fabioz) [inline action_server/docs/CHANGELOG.md]: This changelog entry is actually in the wrong place (it should be in the `unreleased` section).

**Review by fabioz — APPROVED** [review]

**Pull request merged**